abel/direct.py

```diff
@@ -219,9 +219,19 @@ def _pyabel_direct_integral(f, r, correction, int_func=np.trapz):
         f_r = (f[:, 1:] - f[:, :N1-1])/np.diff(r)[None, :]
 
         for i, row in enumerate(f):  # loop over rows (z)
-            out[i, :-1] += I_sqrt[II+1 == JJ]*f_r[i] \
+            corr  = I_sqrt[II+1 == JJ]*f_r[i] \
                     + np.arccosh(r[1:]/r[:-1])*(row[:-1] - f_r[i]*r[:-1])
 
+            
+            if np.isclose(r[0], 0):  # special case for r[0] = 0 
+                corr0 = I_sqrt[II+1 == JJ][0]*f_r[i,0] \
+                        + (row[0] - f_r[i,0]*r[0])
+                out[i, 1:-1] += corr[1:]
+                out[i, 0]    += corr0 
+            else:
+                out[i, :-1] = corr
+            
+
     return out
 
```

Comment:
The original code is fine, together with `r[0]=1.0e-16`
```python
    if correction == 1:
        # computing forward derivative of the data
        f_r = (f[:, 1:] - f[:, :-1])/np.diff(r)[None, :]

        for i, row in enumerate(f):  # loop over rows (z)
            out[i, :-1] += I_sqrt[II+1 == JJ]*f_r[i] \
                    + np.arccosh(r[1:]/r[:-1])*(row[:-1] - f_r[i]*r[:-1])
```

However, this is not a general solution.

Reply:
Yeah, `r[0] = 1.0e-16` avoids returning any `np.nan` values, but if you check the result with the Gaussian case, `out[0]` is pretty far from the correct value. I think that my current code provides a pretty nice result for the gaussian test case, although for poorly understood reasons.

Reply:
Ok. I don't think the exact value of inverse_image[:, 0] is important, since the `direct` method does not provide it, and the real-world value would be noisy, in any case. So, your fix is as good as any. However, the warning message from the evaluation of `np.arccosh(r[1:]/r[:-1])` is not a good look.